astroid/inference.py

```diff
@@ -510,18 +510,28 @@ def _infer_boolop(
 # UnaryOp, BinOp and AugAssign inferences
 
```

Comment:
```suggestion
InferenceResultOrFail = InferenceResult | util.BadUnaryOperationMessage
InferenceFunction = Callable[
        [_T, InferenceContext | None],
        Generator[InferenceResultOrFail, None, None],
    ]

```
Could simplify some of the other typing ?

Reply:
Not sure it makes sense to create an alias for something that is used twice. Most of different combinations of the same thing. We could use a generic, but not sure if that's more readable.

Reply:
There's actually two proposal here, the ``InferenceResultOrFail = InferenceResult | util.BadUnaryOperationMessage`` might be used quite a lot

Reply:
It's only used twice as it is specific to `Unary` operations.

We could make the `Callable` into a alias, but I don't mind it in its current position. The file is quite large anyway.